controllers/connectionTypeController.js

```diff
@@ -41,7 +41,9 @@ exports.setConnectionType = async (req, res, next) => {
 			data: createdConnectionObj,
 		})
 	} catch (error) {
-		next(error)
+		if (error.code !== 11000) next(error)
```

Comment:
if (error.code !== 11000)
this check should not be in the controller but in the service function instead.